users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -776,6 +784,123 @@ def run_scf_baseline():
+        "use_multi_proc_dataset": True,
+    }
+    feature_args = {"class": "ScfNetwork", "size_tf": 256 // 2, "stride_tf": 10 // 2}
+    lr_args = {
+        "peak_lr": 4e-4,
+        "start_lr": 1.325e-05,
+        "end_lr": 1e-5,
+        "increase_epochs": 180,
+        "decrease_epochs": 180,
+        "final_epochs": 0,
+    }
+
+    perturbation_args = [
+        {'speed': {'prob': 0.6, 'minimum': 0.8, 'maximum': 1.2}},
+        {'speed': {'prob': 0.6, 'minimum': 0.7, 'maximum': 1.3}},
+        {'speed': {'prob': 0.5, 'minimum': 0.9, 'maximum': 1.1}},
+        {'speed': {'prob': 0.5, 'minimum': 0.8, 'maximum': 1.2}},
+        {'speed': {'prob': 0.5, 'minimum': 0.7, 'maximum': 1.3}},
+        {'speed': {'prob': 0.4, 'minimum': 0.9, 'maximum': 1.1}},
+        {'speed': {'prob': 0.4, 'minimum': 0.8, 'maximum': 1.2}},
+        {'speed': {'prob': 0.4, 'minimum': 0.7, 'maximum': 1.3}},
+        {'tempo': {'prob': 0.4, 'minimum': 0.9, 'maximum': 1.1}},
+        {'tempo': {'prob': 0.4, 'minimum': 0.8, 'maximum': 1.2}},
+        {'tempo': {'prob': 0.4, 'minimum': 0.7, 'maximum': 1.3}},
+        {'tempo': {'prob': 0.5, 'minimum': 0.9, 'maximum': 1.1}},
+        {'tempo': {'prob': 0.5, 'minimum': 0.8, 'maximum': 1.2}},
+        {'tempo': {'prob': 0.5, 'minimum': 0.7, 'maximum': 1.3}},
+        {'tempo': {'prob': 0.6, 'minimum': 0.9, 'maximum': 1.1}},
+        {'tempo': {'prob': 0.6, 'minimum': 0.8, 'maximum': 1.2}},
+        {'tempo': {'prob': 0.6, 'minimum': 0.7, 'maximum': 1.3}},
+        {'preemphasis': {'prob': 0.9, 'minimum': 0.9, 'maximum': 1.0}},
+        {'preemphasis': {'prob': 0.9, 'minimum': 0.8, 'maximum': 1.0}},
+        {'preemphasis': {'prob': 0.8, 'minimum': 0.9, 'maximum': 1.0}},
+        {'preemphasis': {'prob': 0.8, 'minimum': 0.8, 'maximum': 1.0}},
+        {'codecs': [{'encoding': 'ULAW', 'prob': 0.4}]},
+        {'codecs': [{"encoding": 'ULAW', 'prob': 0.6}]},
+        {'non_linearity': {'prob': 0.4,  'minimum': 0.1, 'maximum': 0.3}},
+        {'non_linearity': {'prob': 0.4,  'minimum': 0.2, 'maximum': 0.4}},
+        {'non_linearity': {'prob': 0.6,  'minimum': 0.1, 'maximum': 0.3}},
+        {'non_linearity': {'prob': 0.6,  'minimum': 0.2, 'maximum': 0.4}},
+    ]
+
+    nn_base_args = {}
+
+    for args in perturbation_args:
+        arg_key = list(args.keys())[0]
+        arg_values = list(args.values())[0]
+
+        # Check if arg_values is a dictionary (has 'minimum' and 'maximum') or a list (like 'codecs')
+        if isinstance(arg_values, dict):
+            key_suffix = f"{arg_key}_{arg_values['prob']}_{arg_values['minimum']}_{arg_values['maximum']}_"
+            report_values = f"{arg_key}: '{arg_values['prob']}_{arg_values['minimum']}_{arg_values['maximum']}'"
+        elif isinstance(arg_values, list):
+            key_suffix = f"{arg_key}_{arg_values[0]['encoding']}_{arg_values[0]['prob']}_"
+            report_values = f"{arg_key} (encoding: {arg_values[0]['encoding']}): '{arg_values[0]['prob']}'"
```

Comment:
This does not work when combining perturbations. You could e.g. have `args =  {'speed': {'prob': 0.6, 'minimum': 0.8, 'maximum': 1.2},  'tempo': {'prob': 0.6, 'minimum': 0.8, 'maximum': 1.2}}`.  You could iterate through the `args` dict and create a name based on specific rules for the keys. E.g.
```
if key in ["speed", "tempo", ...]:
    name += f"{key}_{value["prob"]}_..."
elif key in ["codecs"]:
    name += ...
else:
    raise NotImplementedError
```
Maybe it also helps to create a small function to parse the args into a name. Similarly for the report args (could be done in the same helper function).

Reply:
I made a helper function that generates both